lib/io/src/gpio.rs

```diff
@@ -1,33 +1,47 @@
-/// Abstraction for a GPIO pin so that sensors can be tested with a mock GPIO pin
-pub trait HypedGpioPin {
-    fn is_high(&mut self) -> bool;
-}
-
-pub mod mock_gpio {
-    use heapless::Vec;
-
-    /// A mock GPIO pin that can be used for testing
-    pub struct MockGpio {
-        current_value: bool,
-        next_values: Vec<bool, 10>,
-    }
-
-    impl crate::gpio::HypedGpioPin for MockGpio {
-        fn is_high(&mut self) -> bool {
-            let next_value = self.next_values.pop().unwrap_or(self.current_value);
-            self.current_value = next_value;
-            self.current_value
-        }
-    }
-
-    impl MockGpio {
-        pub fn new(values: Vec<bool, 10>) -> MockGpio {
-            let mut next_values = values.clone();
-            next_values.reverse();
-            MockGpio {
-                current_value: false,
-                next_values,
-            }
-        }
-    }
-}
+/// Abstraction for a GPIO pin so that sensors can be tested with a mock GPIO pin
+pub trait HypedGpioPin {
+    fn is_high(&mut self) -> bool;
+    fn switch_on(&mut self) -> bool;
+    fn switch_off(&mut self) -> bool;
+}

```

Comment:
We actually need to separate the input and output GPIO pins, like in PR #40 [lib/io/src/gpio.rs](https://github.com/Hyp-ed/hyped-2025/pull/40/files#diff-d5bdff5232fdae77ea3da505b80e8189de7dba771accf55b4b245a54d757416e).